server/src/main/java/org/opensearch/indices/replication/PrimaryShardReplicationSource.java

```diff
@@ -78,14 +80,35 @@ public void getSegmentFiles(
     ) {
         final Writeable.Reader<GetSegmentFilesResponse> reader = GetSegmentFilesResponse::new;
         final ActionListener<GetSegmentFilesResponse> responseListener = ActionListener.map(listener, r -> r);
+        // Few of the below assumptions and calculations are added for experimental release of segment replication feature in 2.3
+        // version.These will be changed in next release.
+
+        // Storing the size of files to fetch in bytes.
+        long sizeOfSegmentFiles = 0;
+        for (int i = 0; i < filesToFetch.size(); i++) {
+            sizeOfSegmentFiles += filesToFetch.get(i).length();
+        }
+        // Making sure files size is in correct format to perform time calculation.
+        if(sizeOfSegmentFiles<0){
+            sizeOfSegmentFiles *=-1;
+        }
```

Comment:
Why this change is needed ?

Reply:
So, while testing benchmarks there are scenarios where the size of files to fetch will be in negative. May be something like this "-1239252992" bytes. If size of files to fetch is negative then time value will also be negative, which doesn't makes sense and throws error. I tried to use Math.abs() but that is forbidden so I have no other option. Please suggest if there are any ways to overcome this?

Reply:
This is interesting. Where do you see negative file length value @Rishikesh1159. @mch2 : Do you know when do we have negative file length value ?

Reply:
When running benchmarks I was logging size of files to fetch, I observed that with couple of replication events. I also observed time taken to process both  "-1239252992" bytes size and "1239252992" bytes size is same. So, I was looking for it to convert to a positive value

Reply:
Sorry for confusion. Size of files to fetch will never be in negative. When I was testing locally I was using int type to store the size, so for bigger value int was overflowing and I was negative size of files to fetch in logs. Later I used Long type and this is no more an issue.

Reply:
👍